File: retrofit/src/main/java/webconnect/com/webconnect/WebConnect.java
package webconnect.com.webconnect;

import android.app.Activity;
import android.content.Context;
import android.support.annotation.NonNull;


/**
 * The type Web connect.
 */
public class WebConnect {
    private static volatile WebConnect sWebConnect;

    private WebConnect() {

    }

    /**
     * Get web connect.
     *
     * @return the web connect
     */
    public static WebConnect get() {
        if (sWebConnect == null) {
            synchronized (WebConnect.class) {
                if (sWebConnect == null) {
                    sWebConnect = new WebConnect();
                }
            }
        }
        return sWebConnect;
    }

    /**
     * With builder.
     *
     * @param context the context
     * @param pathSegment     the pathSegment
     * @return the builder
     */
    public static WebParam.Builder with(@NonNull Activity context, @NonNull String pathSegment) {
        return new WebParam.Builder(context, pathSegment);

    }

    /**
     * With builder.
     *
     * @param context the context
     * @param pathSegment     the pathSegment
     * @return the builder
     */
    public static WebParam.Builder with(@NonNull Context context, @NonNull String pathSegment) {
        return new WebParam.Builder(context, pathSegment);
    }

    /*
     * Connect
     */
    @Deprecated
    public static <T> T connect(@NonNull Class<T> cls, @NonNull WebParam webParam) {
        return new RetrofitManager().createService(cls, webParam);
    }
}

File: retrofit/src/main/java/webconnect/com/webconnect/WebConnect.kt
package webconnect.com.webconnect

import android.app.Activity
import android.content.Context


/**
 * The type Web connect.
 */
object WebConnect {
    @Volatile private var sWebConnect: WebConnect? = null

    /**
     * Get web connect.
     *
     * @return the web connect
     */
    fun get(): WebConnect? {
        if (sWebConnect == null) {
            synchronized(WebConnect::class.java) {
                if (sWebConnect == null) {
                    sWebConnect = WebConnect
                }
            }
        }
        return sWebConnect
    }

    /**
     * With builder.
     *
     * @param context the context
     * @param url     the url
     * @return the builder
     */
    fun with(context: Activity, url: String): Builder = Builder(context, url)

    /**
     * With builder.
     *
     * @param context the context
     * @param url     the url
     * @return the builder
     */
    fun with(context: Context, url: String): Builder = Builder(context, url)
}
